src/tls13.c

```diff
@@ -11244,6 +11247,81 @@ const char* wolfSSL_get_cipher_name_by_hash(WOLFSSL* ssl, const char* hash)
 
 
 #ifndef NO_WOLFSSL_SERVER
+#if defined(WOLFSSL_DTLS13) && defined(WOLFSSL_SEND_HRR_COOKIE)
+static void DtlsResetState(WOLFSSL *ssl)
+{
+    /* Reset the state so that we can statelessly await the
+     * ClientHello that contains the cookie. */
+
+    /* Reset DTLS window */
+    w64Zero(&ssl->dtls13Epochs[0].nextSeqNumber);
+    w64Zero(&ssl->dtls13Epochs[0].nextPeerSeqNumber);
+    XMEMSET(ssl->dtls13Epochs[0].window, 0,
+            sizeof(ssl->dtls13Epochs[0].window));
+
+    ssl->keys.dtls_expected_peer_handshake_number = 0;
+    ssl->keys.dtls_handshake_number = 0;
+
+    ssl->msgsReceived.got_client_hello = 0;
+#ifdef WOLFSSL_SEND_HRR_COOKIE
+    /* Remove cookie so that it will get computed again */
+    TLSX_Remove(&ssl->extensions, TLSX_COOKIE, ssl->heap);
+#endif
+
+    /* Reset states */
+    ssl->options.serverState = NULL_STATE;
+    ssl->options.clientState = NULL_STATE;
+    ssl->options.connectState = CONNECT_BEGIN;
+    ssl->options.acceptState  = ACCEPT_BEGIN;
+    ssl->options.handShakeState  = NULL_STATE;
+
+    Dtls13FreeFsmResources(ssl);
+}
+
+static int DtlsAcceptStateless(WOLFSSL *ssl)
+{
+    int ret;
+
+    if (!ssl->options.dtls)
+        return 0;
+
+    switch (ssl->options.acceptState) {
+    case TLS13_ACCEPT_BEGIN:
+
+        while (ssl->options.clientState < CLIENT_HELLO_COMPLETE) {
+            ret = ProcessReply(ssl);
+            if (ret != 0)
+                return ret;
+        }
+
+        if (!IsAtLeastTLSv1_3(ssl->version))
+            return wolfSSL_accept(ssl);
+
+        ssl->options.acceptState = TLS13_ACCEPT_CLIENT_HELLO_DONE;
+        WOLFSSL_MSG("accept state ACCEPT_CLIENT_HELLO_DONE");
+        FALL_THROUGH;
+
+    case TLS13_ACCEPT_CLIENT_HELLO_DONE:
+        if (ssl->options.serverState ==
+                                          SERVER_HELLO_RETRY_REQUEST_COMPLETE) {
+            ret = SendTls13ServerHello(ssl, hello_retry_request);
+            if (ret == 0 || ret == WANT_WRITE)
+                DtlsResetState(ssl);
+            return ret;
+        }
+
+        ssl->options.acceptState = TLS13_ACCEPT_HELLO_RETRY_REQUEST_DONE;
+        WOLFSSL_MSG("accept state ACCEPT_HELLO_RETRY_REQUEST_DONE");
+        FALL_THROUGH;
```

Comment:
Any way to not duplicate the logic here and in `wolfSSL_accept_TLSv13`? Maybe always go here, even for TLS 1.3 (since hrr cookie allows for stateless listening in TLS 1.3 too)?

Reply:
ok, I'll try to refactor that to avoid code duplication

Reply:
After reasoning about it, I would like to avoid that refactoring.
The main problem is that tls1.3 isn't yet completely stateless (it does check the cookie extension only if an HRR was sent in the first place) and I don't want to touch TLS1.3 stuff in this PR. 
I think the amount of code is quite small and this doesn't mess too much with processReply "linear" states.